### PR TITLE
Add support for fetching an individual lazy property with Criteria

### LINKED ISSUE #1861: Lazy loading and Eager initialization for Component
For example:
```csharp
class Organization
{
    public virtual string Name {get;set;}
    public virtual Address Address {get;set;}
}
class Address
{
    public virtual string City {get;set;}
    public virtual Address Street {get;set;}
}

protected OrganizationMap()
{
    Id(p => p.Id).GeneratedBy.UuidHex("N").Length(32);
    Component(p => p.Address, a =>
    {
        a.LazyLoad();
        a.ColumnPrefix("adr_");
        a.Map(p => p.Street);
        a.Map(p => p.City);
    });   
    Map(p => p.Name).Not.Nullable();
}

var orgs = nHsession.QueryOver<Organization>().Fetch(p=>p.Address).Eager.List();
```
Please add feature to Eager load mapped as Component properties

### PR DESCRIPTION
**Pull request by bahusoid**
Follow up to #1922 
To avoid accidental misuse with `SelectMode.FetchLazyProperties` new enum is called `SelectMode.FetchLazyPropertyGroup` (and not `SelectMode.FetchLazyProperty`)

Example of usage:
```C#
var entity = session.QueryOver<Entity>()
	.Fetch(
		SelectMode.FetchLazyPropertyGroup,
		 ec => ec.LazyProp1, ec => ec.LazyProp2, ec => ec.OtherEntity.LazyProp3)
	.Where(ec => ec.Id == id)
	.SingleOrDefault();
```

Fixes #1861

### CODE REVIEW

File: src/NHibernate/Loader/Hql/QueryLoader.cs [hunk 1]
```diff
@@ -119,7 +119,7 @@ protected override bool[] EntityEagerPropertyFetches
 			get { return _entityEagerPropertyFetches; }
 		}
 
-		protected override HashSet<string>[] EntityFetchLazyProperties
+		protected override ISet<string>[] EntityFetchLazyProperties
```
Comment (hazzik) [inline src/NHibernate/Loader/Hql/QueryLoader.cs]: Is this released yet?

File: src/NHibernate/Persister/Entity/AbstractEntityPersister.cs [hunk 2]
```diff
@@ -1641,12 +1641,12 @@ public string PropertySelectFragment(string name, string suffix, bool allPropert
 			return PropertySelectFragment(name, suffix, null, allProperties);
 		}
 
-		public string PropertySelectFragment(string name, string suffix, string[] fetchProperties)
+		public string PropertySelectFragment(string name, string suffix, ICollection<string> fetchProperties)
```
Comment (hazzik) [inline src/NHibernate/Persister/Entity/AbstractEntityPersister.cs]: Is this released yet?

File: src/NHibernate/SelectMode.cs [hunk 3]
```diff
@@ -19,6 +19,12 @@ public enum SelectMode
 		/// Fetch the entity and its lazy properties.
 		/// </summary>
 		FetchLazyProperties,
+		
+		/// <summary>
+		/// Fetch lazy property group
+		/// Note: To fetch single property it must be mapped with unique fetch group (lazy-group)
+		/// </summary>
+		FetchProperty,
```
Comment (hazzik) [inline src/NHibernate/SelectMode.cs]: Should be added to the end.

File: src/NHibernate/Loader/Hql/QueryLoader.cs (hunk 1 above)
Comment (bahusoid) [reply src/NHibernate/Loader/Hql/QueryLoader.cs]: No. All public changes from unreleased #1922

File: src/NHibernate/Persister/Entity/AbstractEntityPersister.cs (hunk 2 above)
Comment (bahusoid) [reply src/NHibernate/Persister/Entity/AbstractEntityPersister.cs]: No. Were added with #1922

**Comment by hazzik** [comment]
>To avoid accidental misuse with `SelectMode.FetchLazyProperties` new enum is called `SelectMode.FetchProperty` (and not `SelectMode.FetchLazyProperty`)

What about changing name of the new value to `FetchLazyPropertyGroup` and old one to `FetchAllLazyProperties` (added "all")?

**Comment by hazzik** [comment]
The commit with proposed changes are here: 6f38e03.

**Comment by bahusoid** [comment]
`FetchLazyPropertyGroup` looks good and it makes it visually distinguishable  with `FetchLazyProperties`. Renaming seems unnecessary to me. But no objections.

**Pull request merged**